docs/references/nextjs/route-protection.mdx

```diff
@@ -0,0 +1,156 @@
+
+The following will require users to be logged in to access the `/dashboard` route, and if they are not it will redirect to sign the sign-in route.
+
+```tsx filename="app/src/middleware.ts"
+import { clerkMiddleware } from '@clerk/nextjs/server';
+
+export default clerkMiddleware(
+  (auth, req) => {
+    if (  req.nextUrl.pathname.startsWith('/dashboard')) {
+      auth().protect();
+    }
+  }
+);
+
+export const config = {
+  matcher: ['/((?!.*\\..*|_next).*)', '/', '/(api|trpc)(.*)'],
+};
+```
+
+## Middleware + layout or page
+
+Applications with more complex protection needs may be well suited to use a mix of Middleware and layout or page based protection. Your application may have a `/dashboard` section, but within that it might have routes or specific pages that are limited to users with a specific role or permission. In a situation such as this you might enforce that only signed in users can access `/dashboard` in your Middleware, and then add specific role based checks the `layout.tsx` for a route like `/dashboard/manage-users`.
+
+### Middleware + layout example
+
+```tsx filename="app/src/middleware.ts"
+import { clerkMiddleware } from '@clerk/nextjs/server';
+
+export default clerkMiddleware(
+  (auth, req) => {
+    if (  req.nextUrl.pathname.startsWith('/dashboard')) {
+      auth().protect();
+    }
+  }
+);
+
+export const config = {
+  matcher: ['/((?!.*\\..*|_next).*)', '/', '/(api|trpc)(.*)'],
+};
+```
+
+```tsx filename="app/src/dashboard/manage-users/layout.tsx" 
+import { auth } from '@clerk/nextjs/server'
+import { redirect } from 'next/navigation'
+
+export default async function HookLayout({ children }: { children: React.ReactNode }) {
+  if (!auth().has({ role: "org:admin" })) {
+    redirect("/")
+  }
+  return <>{children}</>
+}
+```
+
+### Middleware + page example
+
+```tsx filename="app/src/middleware.ts"
+import { clerkMiddleware } from '@clerk/nextjs/server';
+
+export default clerkMiddleware(
+  (auth, req) => {
+    if (  req.nextUrl.pathname.startsWith('/dashboard')) {
+      auth().protect();
+    }
+  }
+);
+
+export const config = {
+  matcher: ['/((?!.*\\..*|_next).*)', '/', '/(api|trpc)(.*)'],
+};
+```
+
+```tsx filename="app/src/dashboard/manage-users/page.tsx" 
+import { auth } from '@clerk/nextjs/server'
+
+export default function Page() {
+
+if (!auth().has({ role: "org:admin" })) {
+    return <p>You do not have permission to view this page.</p>
+  }
+
+  return (
+    <>
+      <h1>Manage Users</h1>
+      <p>Manage your users, invite new users or ban users</p>
+    </>
+  );
+}
+```
+
+## Layout or page only
+
+Clerk's `clerkMiddleware()` is purely opt-in, meaning if ou don't add any specific protection to the Middleware all routes are public. This allows you to then enforce protections via only layouts or even pages and keep your protection logic scoped to that specific area it govens in your application.
```

Comment:
```suggestion
Since Clerk's `clerkMiddleware()` is purely opt-in, meaning if you don't add any specific protection to the Middleware all routes are by default public. This allows you to then enforce protections via only layouts or even pages and keep your protection logic scoped to that specific area it governs, in your application.
```